lib/pbench/server/api/__init__.py

```diff
@@ -380,6 +392,11 @@ def post(self):
                     exc,
                 )
             abort(400, message=message)
+
+        # Write the checksum as .md5 file
+        with open(f"{full_path}.md5", "w") as f:
+            f.write(f"{md5sum}  {filename}\n")
```

Comment:
We have have settled on in the exist SSH based protocol, and it seems it would apply here as well, is that the following steps are taken to ensure we never see a tar ball without its `.md5` file:

1. Write `.tar.xz` file to a hidden temporary file (see [Tool Data Sink, line 744](https://github.com/distributed-system-analysis/pbench/blob/master/lib/pbench/agent/tool_data_sink.py#L744)) in the same directory it will eventually live
2. After all bits have been received and written to disk, verify the expected MD5 value matches header
3. Write MD5 value to `.tar.xz.md5` first
  - On failure to write, return a 500 error, temporary hidden file will evaporate
4. Now that we have successfully written the `.tar.xz.md5` file, we can `link()` the hidden temporary file to its final name
  - See Tool Data Sink line [787](https://github.com/distributed-system-analysis/pbench/blob/master/lib/pbench/agent/tool_data_sink.py#L787)

Reply:
Ahh yeah writing data to tmpfile first is better